agent/command/git.go

```diff
@@ -384,14 +384,15 @@ func (c *gitFetchProject) buildModuleCloneCommand(conf *internal.TaskConfig, opt
 }
 
 func (c *gitFetchProject) opts(projectMethod, projectToken string, conf *internal.TaskConfig) (cloneOpts, error) {
+	shallowCloneEnabled := conf.Distro != nil && !conf.Distro.DisableShallowClone
```

Comment:
I don't have any context on why some distros would want to have shallow clone enabled/disabled. It seems like we should respect the distro-level shallow clone setting but for container tasks, we should always allow shallow cloning and assume that if users ask for a shallow clone, they'll supply an image that supports shallow cloning.

Reply:
Gotcha, updated the bool so that `shallowCloneEnabled` will be true if Distro is nil